src/uthash.h

```diff
@@ -26,9 +26,11 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #define UTHASH_VERSION 2.0.1
 
+#ifndef UTHASH_FREESTANDING_IMPLEMENTATION
 #include <string.h>   /* memcmp,strlen */
 #include <stddef.h>   /* ptrdiff_t */
 #include <stdlib.h>   /* exit() */
```

Comment:
You get NULL from somewhere.

As an alternative to providing configuration macros for all of `XXX_t`s, uthash may simply demand that they are present; and list all things it demands are present in free standing configuration.

That statement about you not taking a patch makes sense. I consider `NULL` a standard (something that better be defined somehow), but `XXX_t` not so standard, and instead of requiring the caller to provide them, giving the caller a way to hook. As long as it's done in one way or another, I'm actually fine (for embedding), I just don't know what the litmus test for those things is. A developer may want to override `malloc()` because they want a different memory allocation, but it also may be because they have no `malloc()`. May be the litmus test should be: _"Is it conceivable that THING exists, but does/means something else that the developer might want it to"_. Like, overriding a `malloc()`, and for uthash only, is possible, but just a pain. Having uuin32_t that doesn't mean "unsigned 32-bit integer" is not conceivable.